recipe/meta.yaml

```
@@ -11,30 +11,47 @@ source:
   sha256: "{{ sha256 }}"
 
 build:
+  # No mongodb on s390x
+  skip: True  # [py<34 or s390x]
   number: 0
-  noarch: python
-  script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed -vvv "
+  script: {{ PYTHON }} -m pip install . --no-deps --ignore-installed --no-build-isolation -vv
 
 requirements:
   host:
     - pip
     - python
+    - setuptools
+    - wheel
   run:
     - python
 
 test:
+  source_files:
+    - tests
   imports:
     - mockupdb
   requires:
-    - pymongo >=3
+    - pymongo >=3,<4  # [not (py>311 or (py>310 and osx and arm64) )]
+    # XXX even though we're not running the tests for py>311, we need
+    # *a* mongodb, which has a vendored bson, which mockupdb requires
+    # (for pip check).
+    - pymongo >=3     # [py>311 or (py>=310 and (osx and arm64))]
```

```suggestion
```
You could remove all this and move the comment above pytest.

---

I think I still need all of that as the tests are going to either use functions removed `pymongo` `4.0` or are going to need _a_ `pymongo` 3 just so it can satisfy the `pip check` which will want the vendored `bson`.

I don't like it but it seems necessary to get some tests running.

---

@cbouss ?

---

`pip check` should work using just the `run`time dependencies. If we need to add other dependencies in the `test` section just to make the `pip check` pass, that's not what our users will do usually. What's the problem with `pip check` and `bson`?

---

```suggestion
    - pymongo >=3,<4  # [not (py>311 or (py>310 and osx and arm64) )]
```

wouldn't this be enough?

---

The `bson` statement was because I forgot to add `pymongo>=3` to the `requirements.run` (schoolboy error!) and then didn't clean up the commentary.

After that, the additional `<4` constraint, only for those platforms that are going to run the tests _is_ enough.

Hopefully, I've cleaned that up.

---

All clear! Thanks for the addition!